spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -561,14 +561,11 @@ public Scan buildChangelogScan() {
 
     boolean emptyScan = false;
     if (startTimestamp != null) {
-      if (table.currentSnapshot() != null
-          && table.currentSnapshot().timestampMillis() < startTimestamp) {
+      if (table.currentSnapshot() == null
+          || table.currentSnapshot().timestampMillis() < startTimestamp) {
```

Comment:
Nit, but for me this is easier to reason. When `startTimestamp` is after the timestamp of the current table snapshot:
```suggestion
          || startTimestamp > table.currentSnapshot().timestampMillis()) {
```

Reply:
Thanks for the comment!
I just updated condition part and added the changes to Spark 3.3, 3.4. Feel free to take a look when you have time.